packages/salesforcedx-vscode-core/src/commands/projectDeployStart.ts

```diff
@@ -5,235 +5,217 @@
-    const table = new Table();
-    const titleType = this.getDeployType();
-
-    const successes = parser.getSuccesses();
-    const errors = parser.getErrors();
-    const pushedSource = successes ? successes.result.files : undefined;
-    if (pushedSource || parser.hasConflicts()) {
-      const rows = pushedSource ?? errors?.files;
-      const title = !parser.hasConflicts() ? nls.localize(`table_title_${titleType}ed_source`) : undefined;
-      const outputTable = this.getOutputTable(table, rows, title);
-      if (parser.hasConflicts()) {
-        channelService.appendLine(`${nls.localize('push_conflicts_error')}\n`);
-      }
-      channelService.appendLine(outputTable);
-      if (pushedSource && pushedSource.length === 0) {
-        const noResults = `${nls.localize('table_no_results_found')}\n`;
-        channelService.appendLine(noResults);
+  private async checkConflictsForChangedFiles(): Promise<boolean> {
+    try {
+      const messages = getConflictMessagesFor('deploy_with_sourcepath');
+      if (!messages) {
+        return true; // No conflict messages available, continue
       }
-    }
 
-    if (errors && !parser.hasConflicts()) {
-      const { name, message, files } = errors;
-      if (files) {
-        const outputTable = this.getErrorTable(table, files, titleType);
-        channelService.appendLine(outputTable);
-      } else if (name && message) {
-        channelService.appendLine(`${name}: ${message}\n`);
-      } else {
-        console.log(`There were errors parsing the push operation response.  Raw response: ${JSON.stringify(errors)}`);
+      // Show channel output and log conflict detection start once for the entire operation
+      channelService.showChannelOutput();
+      channelService.showCommandWithTimestamp(
+        `${nls.localize('channel_starting_message')}${nls.localize('conflict_detect_execution_name')}\n`
+      );
+
+      const { username } = WorkspaceContext.getInstance();
+      if (!username) {
+        const errorMsg = nls.localize('conflict_detect_no_target_org');
+        channelService.appendLine(errorMsg);
+        return false;
       }
-    }
-  }
 
-  protected getOutputTable(
-    table: Table,
-    rows: ProjectDeployStartResult[] | undefined,
-    outputTableTitle: string | undefined
-  ) {
-    const outputTable = table.createTable(
-      rows ?? [],
-      [
-        { key: 'state', label: nls.localize('table_header_state') },
-        { key: 'fullName', label: nls.localize('table_header_full_name') },
-        { key: 'type', label: nls.localize('table_header_type') },
-        { key: 'filePath', label: nls.localize('table_header_project_path') }
-      ],
-      outputTableTitle
-    );
-    return outputTable;
-  }
+      // Create a single TimestampConflictChecker instance for all files
+      const timestampChecker = new TimestampConflictChecker(false, messages, true);
+
+      // Check conflicts for each changed file
+      for (const filePath of this.changedFilePaths) {
```

Comment:
@CristiCanizales this for loop and the way timestampChecker.checkFileWithoutLogging is implemented is going to a retrieve of the project metadata from the target for each changed file.